feature/platform/tests/telemetry_inventory_test/telemetry_inventory_test.go

```diff
@@ -770,6 +788,32 @@ func ValidateComponentState(t *testing.T, dut *ondatra.DUTDevice, cards []*oc.Co
 	}
 }
 
+func findParent(t *testing.T, dut *ondatra.DUTDevice, cName string, cur string) (parent string, parentType oc.Component_Type_Union, present bool) {
+	val := gnmi.Lookup(t, dut, gnmi.OC().Component(cur).Parent().State())
+	parent, ok := val.Val()
+	if !ok {
+		t.Errorf("Component %s Parent component NOT found in the hierarchy tree of component", cName)
+	}
+	pLoookup := gnmi.Lookup(t, dut, gnmi.OC().Component(parent).Type().State())
+	parentType, present = pLoookup.Val()
+	return parent, parentType, present
+}
+
+func fanParent(t *testing.T, dut *ondatra.DUTDevice, cName string, cur string) {
+	_, parentType, present := findParent(t, dut, cName, cur)
+	if !present {
+		t.Errorf("Component %s Parent: Parent component NOT found in the hierarchy tree of fan component", cName)
+	}
+	switch parentType {
```

Comment:
I know that a FAN's parent can be a FAN_TRAY (for example in a Juniper PTX10008), but I am not certain if there are scenarios in which the FAN's parent is the CHASSIS (i.e., if fans can exist not in a tray). I am assuming this is a possibility.

Reply:
AFAIR the `parent` check is actually an `ancestor` check that does a recursive validation. the expectation is that all components have `chassis` ancestor. the immediate parent is usually not important and does not require a validation (unless something has changed recently).

Reply:
In https://github.com/openconfig/public/pull/1095 we added new leaves for install-position and install-component to get better location data for components. This means the ancestor that the component is installed in (which sometimes is the immediate parent, for example a transceiver's install-component would be the immediate parent port component) is important.

For the fan component, I would expect the install-position to be "1" as an example, and the install-component (and for now also state/parent) to be the "FAN_TRAY" because this is what it is installed in.

Reply:
Some examples:

For a Nokia 7250-ixr10e, the component with type "FAN" has `state/parent` of "Chassis", and there are no fan tray components on this device.

For a Juniper PTX10008, the component with type "FAN" has `state/parent` of "Fan Tray 0" (there are fan tray components on this device that right now have type "FRU").

Reply:
`parent` and `install-component` may be different values.   

IMO this check could just verify that a component of type `FAN` traces recursively to CHASSIS (using the `findParent` function).  Even if a FAN_TRAY is present in the tree, I don't think we can require FAN_TRAY to always be a parent of a `FAN`.  A device might even have some FANs which are in a FAN_TRAY and others which are not.  

So I vote to use `findParent` to recursively trace to`CHASSIS` instead of a fan specific parent validation.

Reply:
Ok that makes sense, thanks for the clarification!

I've removed the fan-specific checks on the immediate parent type, so now the parent validation for fan components is the same as for all other types (i.e., that it has an ancestor of type CHASSIS).